uni/lib/controller/cleanup.dart

```diff
@@ -53,26 +49,24 @@ Future<void> cleanupCachedFiles() async {
 
   final toCleanDirectory = await getApplicationDocumentsDirectory();
   final threshold = DateTime.now().subtract(const Duration(days: 30));
-  final directories = toCleanDirectory.listSync(followLinks: false);
 
-  for (final directory in directories) {
-    if (directory is Directory) {
-      final files = directory.listSync(recursive: true, followLinks: false);
+  await cleanDirectory(toCleanDirectory, threshold);
 
-      final oldFiles = files.where((file) {
-        try {
-          final fileDate = File(file.path).lastModifiedSync();
-          return fileDate.isBefore(threshold);
-        } catch (e) {
-          return false;
-        }
-      });
+  await PreferencesController.setLastCleanUpDate(DateTime.now());
+}
 
-      for (final file in oldFiles) {
-        await File(file.path).delete();
-      }
+Future<void> cleanDirectory(Directory directory, DateTime threshold) async {
+  final entities = directory.listSync(recursive: true, followLinks: false);
+  final toDeleteEntities = entities.whereType<File>().where((file) {
+    try {
+      final fileDate = File(file.path).lastModifiedSync();
```

Comment:
```suggestion
      final fileDate = file.lastModifiedSync();
```
The cast is not needed